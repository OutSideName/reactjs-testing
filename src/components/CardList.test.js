import {shallow} from 'enzyme';
import React from 'react';
import CardList from './CardList';
import renderer from 'react-test-renderer';

it("testing render json", ()=>{
  const wrapper = renderer.create(<CardList />).toJSON();
  expect(wrapper).toMatchSnapshot();
})

it("testing instance", ()=> {
  const wrapper = shallow(<CardList />);
  expect(wrapper.instance()).toMatchSnapshot();
})



