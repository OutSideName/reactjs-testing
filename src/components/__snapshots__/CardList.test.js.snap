// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing instance 1`] = `
CardList {
  "cardComponent": [Function],
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <CardList />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div />,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`testing render json 1`] = `<div />`;
